versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -1060,15 +1084,15 @@ protected Hash copyCommits(
               newKeyLists);
       keyListDistance = newEntry.getKeyListDistance();
 
-      if (!newEntry.getHash().equals(sourceCommit.getHash())) {
+      if (!newEntry.getHash().equals(sourceCommitValueHash)) {
         commitsChronological.set(i, newEntry);
+        targetHead = newEntry.getHash();
       } else {
         // Newly built CommitLogEntry is equal to the CommitLogEntry to transplant.
         // This can happen, if the commit to transplant has NO_ANCESTOR as its parent.
         commitsChronological.remove(i);
+        targetHead = sourceCommit.getHash();
```

Comment:
This assignment is very dangerous and can lead to data corruption.
The copied commit _will be deleted_ if the transplant/merge ran into an issue later. With this assignment, you tell it to delete the source-commit, which must not happen.

Reply:
IMO this part of the code is for cases, when we choose "not to copy" the commits. As per my understanding, the else block applies to cases, when sourceCommit and newEntry arrive at exactly same definition. The if condition prior to this change can be a good ref - `newEntry.getHash().equals(sourceCommit.getHash())` 

With the commitUpdate function is triggered, the newEntry uses updated meta and the computed hash value can be different from source commit hash. To retain the behaviour, we are matching the commit definition by checking a recomputed source hash.
In this particular case, the updated commit metadata isn't saved anywhere because copy didn't happen. Hence, we point it back to sourceCommit's hash as it will be used as the targetHead.

> The copied commit will be deleted if the transplant/merge ran into an issue later. With this assignment, you tell it to delete the source-commit, which must not happen.
Looking at the code, I'm not able to visualise any scenario where a source commit would be deleted. Can you please elaborate.

Reply:
Well, I mentioned this earlier - 

> @snazy @nastra Changing the commit metadata is causing a deviation in the hash calculation here - [AbstractDatabaseAdapter.java#L581](https://github.com/projectnessie/nessie/blob/20d6ec68b392382ada33386b7661c0b52d1f71d3/versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java#L581)

newEntry is now constructed with an updated commitMeta, hence it'll have a different hash than source commit.
> With the commitUpdate function is triggered, the newEntry uses updated meta and the computed hash value can be different from source commit hash.

To retain the definition match, we're evaluating - ( `updateCommitMeta(commitsChronological.get(i))::hash` against `newEntry.getHash()`.  If copy isn't required, `updateCommitMeta` isn't required either, and targetHead is set to `commitsChronological.get(i).getHash()`

Reply:
`sourceCommit.getHash()` is the original hash whereas `sourceCommitValueHash` is computed from the object values. I can create a new variable `updatedSourceCommit` if that would reduce the confusion.

The `targetHead` has to be set to the original commit right? `sourceCommitValueHash/newEntry.getHash()` aren't representing the same.